### DISCUSSION TITLE
App Launcher script not populating the apps.json properly

### DISCUSSION THREAD

**Discussion by horacioh** [johnlindquist/kit#379] [Errors] [answered]
https://user-images.githubusercontent.com/725120/127755100-7818c6db-355b-40af-b3be-477311a40031.mov

For some reason the App launcher is not populating properly the `apps.json` file in the db.

- current version: 1.3.0-beta.34

Started the bug report [here](https://twitter.com/hhg2288/status/1421602196257394689)

anything else you need me to try please let me know!

**Comment by johnlindquist** [comment]
Could you try deleting `~/.kenv/db/apps.json` again, then run this in your terminal:

(this runs the exact same script, just terminal based. If this works, it's something wrong with how the app is running the script. Maybe a permissions issue...)

```bash
~/.kenv/bin/app-launcher
```

If that doesn't work, please try running this in your terminal:

(`fileSearch` is a wrapper around the `mdfind` tool on osx. It runs this command internally to find applications):

```bash
mdfind -onlyin / "kind:application"
```

Do either of those do anything strange?

**Reply by horacioh** [reply]
this is the result of both:

```bash
~
→ ~/.kenv/bin/app-launcher
Error: At least one choice must be selectable
    at AutoComplete.reset (/Users/horacio/.kit/node_modules/enquirer/lib/types/array.js:38:13)
    at async AutoComplete.initialize (/Users/horacio/.kit/node_modules/enquirer/lib/types/array.js:25:5)
    at async /Users/horacio/.kit/node_modules/enquirer/lib/prompt.js:236:7

~
→ mdfind -onlyin / "kind:application"

~
→
```

**Reply by johnlindquist** [reply]
Looks like it's "mdfind" then...

Can you try a few mdfind commands and see if anything works?

https://metaredux.com/posts/2019/12/22/mdfind.html

**Reply by johnlindquist** [reply]
I'm curious if "kind:application" would be different for each language 🤔

**Reply by horacioh** [reply]
seems to be some sort of error at Spotlight level. I'm not even able to have any autocomplete in Spotlight. I think I disable spotlight indexes and now I dunno how to enable it 🤷🏼‍♂️.

is there a way to create the cached list from another thing that is not spotlight?

I dunno if it will be different in other language, I have my machine in english :)

**Reply by johnlindquist** [reply]
Yeah, we can readdir on "/Applications". We'd probably have to recurse and filter based on files ending in ".app" or something to make sure we find everything. It would be a fun exercise. The spotlight cache just makes that all easy since it tags everything.

Someday next year, I'll probably write tools (probably in Rust) to make our own custom file index for Script Kit so we have complete control.

**Comment by horacioh** [comment, marked as answer]
OK, the problem was that my Spotlight cached data was corrupted (incomplete). by re-Indexing all the Spotlight data I was able to get App results from it!.

This was an Issue with my Soptlight index data and nothing related to ScriptKit!
If you want to know how I rebuild my Soptlight index, [this Alfred troubleshooting](https://www.alfredapp.com/help/troubleshooting/indexing/#rebuild) helped me a lot!

Basically what you can do is:
- Run this two commands to Disable and Re-enable Spotlight indexes:
```bash
sudo mdutil -a -i off
sudo mdutil -a -i on
```
- remove `~/.kenv/db/apps.json`
- run the App Launcher Script again and it should work!

**Reply by johnlindquist** [reply]
Thx @horacioh !
